### ISSUE TITLE
Can't edit server address if matrix.org is not reachable

### ISSUE THREAD

**Issue by licaon-kter** [vector-im/vector-android#393]
You basically get stuck waiting:

* Start app
* Get loading circle animation that blocks interaction
* After 15 seconds it timeouts, shows a toast
* You can interact with the advanced option down arrow for about 1 second
* Get loading circle animation again 
* After 15 seconds it timeouts, shows a toast again
* You can interact with the server fields for about 1 second
* Repeat...

Server status should not be checked until the user decides to: register or login.

**Comment by azrdev** [comment]
I just experienced this same issue, with some difference:

> You can interact with the advanced option down arrow for about 1 second

actually, after the 15s (?) timeout I could interact with the ui freely until
* I finished/paused (not sure) editing the home server field or
* the screen autolocked and then got back to riot after unlock

> Server status should not be checked until the user decides to: register or login.

I think it should be checked, but the progress-circle must not block the rest of the UI

**Comment by licaon-kter** [comment]
Yes, sometimes stays a bit longer than 1 sec, but still glitchy.

**Comment by licaon-kter** [comment]
Tested a dev build again:
* tries to connect 15s
* timeouts
* you can check the server checkbox
* tries to connect 15s
* timeouts
* you can edit servers now
